src/EnergyPlus/ElectricBaseboardRadiator.cc

```diff
@@ -872,31 +770,52 @@ namespace ElectricBaseboardRadiator {
 				// should include this.
 				LoadMet = ( SumHATsurf( ZoneNum ) - ZeroSourceSumHATsurf( ZoneNum ) ) + ( QBBCap * ElecBaseboard( BaseboardNum ).FracConvect ) + ( RadHeat * ElecBaseboard( BaseboardNum ).FracDistribPerson );
 
-				if ( LoadMet < 0.0 ) { // No cooling output
-					AirOutletTemp = AirInletTemp;
-					ElecBaseboard( BaseboardNum ).ElecUseRate = 0.0;
+				if ( LoadMet < 0.0 ) {
+					// This basically means that SumHATsurf is LESS than ZeroSourceSumHATsurf which
+					// should not happen unless something unusual is happening like a fast change
+					// in temperature or some sort of change in internal load.  This is not a problem
+					// normally, but when LoadMet goes negative the choice is to either zero out
+					// the baseboard or give it another shot at getting an accurate reading on
+					// what is happening in the zone.  If it is still predicting a negative heating
+					// load, then zero everything out.
+					// First, turn off the baseboard:
+					Real64 TempZeroSourceSumHATsurf;
+					QBBElecRadSource( BaseboardNum ) = 0.0;
+					DistributeBBElecRadGains();
+					HeatBalanceSurfaceManager::CalcHeatBalanceOutsideSurf( ZoneNum );
+					HeatBalanceSurfaceManager::CalcHeatBalanceInsideSurf( ZoneNum );
+					TempZeroSourceSumHATsurf = SumHATsurf( ZoneNum );
+					// Now, turn it back on:
+					QBBElecRadSource( BaseboardNum ) = RadHeat;
+					DistributeBBElecRadGains();
+					HeatBalanceSurfaceManager::CalcHeatBalanceOutsideSurf( ZoneNum );
+					HeatBalanceSurfaceManager::CalcHeatBalanceInsideSurf( ZoneNum );
+					// Recalculate LoadMet with new ZeroSource... term and see if it is positive now.  If not, shut it down.
+					LoadMet = ( SumHATsurf( ZoneNum ) - TempZeroSourceSumHATsurf ) + ( QBBCap * ElecBaseboard( BaseboardNum ).FracConvect ) + ( RadHeat * ElecBaseboard( BaseboardNum ).FracDistribPerson );
+					if ( LoadMet < 0.0 ) {
```

Comment:
Since I don't understand this I'll ask. What does it mean when TempZeroSourceSumHATsurf is different from ZeroSourceSumHATsurf( ZoneNum ) calculated at line 569 when FirstHVACIteration = true ? To me that means some model somewhere in code did not correctly account for the boundary condition that was present at the beginning of the time step (an energy leak).

Reply:
If there were a fast change in temperature, I assume from some other HVAC model running, wouldn't QZnReq reflect that capacity and update the load the baseboard needed to meet accordingly?

    QZnReq = ZoneSysEnergyDemand( ZoneNum ).RemainingOutputReqToHeatSP;

And isn't internal loads accounted for in predictor?

Reply:
I just wanted to think through this one more time. If you are comfortable then that's good enough for me.

Reply:
@rraustad Totally understand.  It's one of those weird things with any of the systems that have a radiant component to them.  It's the best way that I know to get some sort of estimate as to what effect that radiant component will have on meeting the load.  It's not perfect but it avoids adding yet another layer of iteration in a section of code that has already been critiqued in the past as being slow.  So, it's a compromise but I am comfortable with it for now.